token/core/common/wcache.go

```diff
@@ -36,13 +39,13 @@ func NewWalletIdentityCache(backed WalletIdentityCacheBackendFunc, size int) *Wa
 func (c *WalletIdentityCache) Identity() (view.Identity, error) {
 	select {
 	case entry := <-c.ch:
-		if logger.IsEnabledFor(zapcore.DebugLevel) {
-			logger.Debugf("fetch identity from producer channel done [%s][%d]", entry)
+		if c.Logger.IsEnabledFor(zapcore.DebugLevel) {
+			c.Logger.Debugf("fetch identity from producer channel done [%s][%d]", entry)
```

Comment:
No need for `IsEnabledFor` for none of the two cases. Also there seems to be an extra argument needed `[%d]`